File: container_with_most_water.js
const maxArea = function(array) {
    //write code here
    let left = 0;
    let right = array.length - 1;

    let maxArea = 0;

    while (left < right) {
        let area = Math.min(array[left], array[right]) * (right - left);
        maxArea = Math.max(area, maxArea);

        if (array[left] < array[right]) {
            left += 1
        } else {
            right -= 1
        }
    }

    return maxArea

}

File: container_with_most_water.py
def max_area(array):
    left=0
    right=len(array)-1
    
    max_area=0
    
    while(left<right):
        area=min(array[left],array[right])*(right-left)
        max_area = max(area,max_area)
        
        if array[left]<array[right]:
            left +=1 
        else:
            right -=1
    
    return max_area